webapp/app.py

```diff
@@ -0,0 +1,152 @@
+from flask import Flask, render_template, request, send_file, Response
+from werkzeug import FileStorage
+from zipfile import ZipFile
+import uuid
+import os
+import shutil
+from music21 import *
+import json
+from typing import List
+from inspect import getsourcefile
+# import from parent dir according to https://stackoverflow.com/a/9331002
+import os.path as path, sys
+current_dir = path.dirname(path.abspath(getsourcefile(lambda:0)))
+sys.path.insert(0, current_dir[:current_dir.rfind(path.sep)])
+import midi_to_part_mp3s
+sys.path.pop(0)
+
+
+app = Flask(__name__)
+
+FILE: str = 'file'
+SOPRANO: str = 'soprano'
+ALTO: str = 'alto'
+TENOR: str = 'tenor'
+BASS: str = 'bass'
+INSTRUMENT: str = 'instrument'
+MUSICXML: str = 'musicxml'
+OUTPUT: str = 'output'
+
+TEMP_FOLDER: str = 'tmp'
+OUTPUT_FOLDER: str = 'out'
+FILENAME: str = 'part-mp3s.zip'
+
+
+@app.route("/")
+def index():
+    return render_template('index.html')
+
+
+@app.route("/midi-info", methods=['POST'])
+def midi_info():
+    random_folder_name = _check_tmp_folder()
+    midi_file_name = _get_musicxml_file(random_folder_name)
+    print(midi_file_name)
+    score: stream.Score = converter.parse(midi_file_name)
+    print(score)
+    musicxml_filename = "%s/%s" % (random_folder_name, 'score.musicxml')
+    score.write(fp=musicxml_filename)
+    xml_content = None
+    with open(musicxml_filename) as file_stream:
+        xml_content = file_stream.read()
+    response_dict = {}
+    response_dict['musicxml'] = xml_content
+    response_dict['parts'] = []
+    part_stream = score.parts.stream()
+    for part in part_stream:
+        response_dict['parts'].append(part.id)
+    return json.dumps(response_dict)
+
+
+@app.route("/download", methods=['POST'])
+def midi_to_mp3():
```

Comment:
I know this is only a version 1 so take this as a request for a subsequent PR:  These methods are far too long and should be split up. For example (see next comment)